festim/boundary_conditions/sieverts_bc.py

```diff
@@ -0,0 +1,58 @@
+import festim as F
+from dolfinx import fem
+import ufl
+
+
+def siverts_law(T, S_0, E_S, pressure):
+    """Applies the Sieverts law to compute the concentration at the boundary"""
+    S = S_0 * ufl.exp(-E_S / F.k_B / T)
+    return S * pressure**0.5
+
+
+class SievertsBC(F.DirichletBC):
+    """
+    Dirichlet boundary condition class
+
+    Args:
+        subdomain (festim.Subdomain): the subdomain where the boundary
+            condition is applied
+        value (float or fem.Constant): the value of the boundary condition
+        species (str): the name of the species
+
+    attributes:
+        subdomain (festim.Subdomain): the subdomain where the boundary
+            condition is applied
+        value (float or fem.Constant): the value of the boundary condition
+        species (str): the name of the species
+    """
+
+    def __init__(self, subdomain, S_0, E_S, pressure, species) -> None:
+        super().__init__(value=None, species=species, subdomain=subdomain)
+        self.subdomain = subdomain
+        self.S_0 = S_0
+        self.E_S = E_S
+        self.pressure = pressure
+        self.species = species
```

Comment:
No need to assign the subdomain and species attribute here

```suggestion
        self.S_0 = S_0
        self.E_S = E_S
        self.pressure = pressure
```